### PR TITLE
Fix chapter recognition regex and detail number

### PR DESCRIPTION
**Pull request by CarlosEsco**
Wasnts matching on  vol. 1 ch. 10 previously so mangadex last chapter was showing volume number.

Changed details page to show blank instead of -1 or 0 for manga without chapters or with chapters but has no numbers like one shots.
Old  
image

New
image

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/util/ChapterRecognition.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ object ChapterRecognition {
      * All cases with Ch.xx
      * Mokushiroku Alice Vol.1 Ch. 4: Misrepresentation -R> 4
      */
-    private val basic = Regex("""(?<=ch\.)([0-9]+)(\.[0-9]+)?(\.?[a-z]+)?""")
+    private val basic = Regex("""(?<=ch\.) *([0-9]+)(\.[0-9]+)?(\.?[a-z]+)?""")
```
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/util/ChapterRecognition.kt]: Add test case for the new format

File: app/src/main/java/eu/kanade/tachiyomi/util/ChapterRecognition.kt [hunk 2]
```diff
@@ -37,6 +37,7 @@ object ChapterRecognition {
      */
     private val unwantedWhiteSpace = Regex("""(\s)(extra|special|omake)""")
 
+
```
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/util/ChapterRecognition.kt]: I know Material Design is all about white space :), But perhaps it's not needed

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoController.kt [hunk 3]
```diff
@@ -249,10 +248,12 @@ class MangaInfoController : NucleusController<MangaInfoPresenter>(),
      * @param count number of chapters.
      */
     fun setChapterCount(count: Float) {
-        manga_chapters?.text = DecimalFormat("#.#").format(count)
+        if (count > 0f) {
+            manga_chapters?.text = DecimalFormat("#.#").format(count)
+        }
```
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoController.kt]: Add a else case which outputs something like "Unkown"

Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoController.kt]: Good idea will implement

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoController.kt [hunk 4]
```diff
@@ -249,10 +248,14 @@ class MangaInfoController : NucleusController<MangaInfoPresenter>(),
      * @param count number of chapters.
      */
     fun setChapterCount(count: Float) {
-        manga_chapters?.text = DecimalFormat("#.#").format(count)
+        if (count > 0f) {
+            manga_chapters?.text = DecimalFormat("#.#").format(count)
+        } else {
+            manga_chapters?.text = resources!!.getString(R.string.unknown)
```
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoController.kt]: use ?! not !!

Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoController.kt]: done

**Comment by NoodleMage** [comment]
lol. What happened to your pull? xD

image

**Comment by CarlosEsco** [comment]
What i get for trying to switch to LF.  Let me fix

**Pull request merged**